### PR TITLE
[REVIEW ONLY] JAMES-2578 Add utils for Mailet new API & apache-mailet-standard warnings

### PR DESCRIPTION
**Pull request by blackheaven**
See #1883

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[120af7c3194ea7f36be2a09ba78a2ebf963f778d] [INFO] Some messages have been simplified; recompile with -Xdiags:verbose to get full output
[120af7c3194ea7f36be2a09ba78a2ebf963f778d] [INFO] -------------------------------------------------------------
[120af7c3194ea7f36be2a09ba78a2ebf963f778d] [ERROR] COMPILATION ERROR : 
[120af7c3194ea7f36be2a09ba78a2ebf963f778d] [INFO] -------------------------------------------------------------
[120af7c3194ea7f36be2a09ba78a2ebf963f778d] [ERROR] /james-project/mailet/standard/src/main/java/org/apache/james/transport/matchers/SMTPAuthUserIs.java:[62,51] method getValueAndCastFromMail in class org.apache.mailet.AttributeUtils cannot be applied to given types;
[120af7c3194ea7f36be2a09ba78a2ebf963f778d]   required: org.apache.mailet.Mail,org.apache.mailet.AttributeName,java.lang.Class<T>
[120af7c3194ea7f36be2a09ba78a2ebf963f778d]   found: org.apache.mailet.Mail,java.lang.String,java.lang.Class<java.lang.String>
[120af7c3194ea7f36be2a09ba78a2ebf963f778d]   reason: cannot infer type-variable(s) T
[120af7c3194ea7f36be2a09ba78a2ebf963f778d]     (argument mismatch; java.lang.String cannot be converted to org.apache.mailet.AttributeName)
[120af7c3194ea7f36be2a09ba78a2ebf963f778d] [ERROR] /james-project/mailet/standard/src/main/java/org/apache/james/transport/matchers/HasException.java:[71,17] method getValueAndCastFromMail in class org.apache.mailet.AttributeUtils cannot be applied to given types;
[120af7c3194ea7f36be2a09ba78a2ebf963f778d]   required: org.apache.mailet.Mail,org.apache.mailet.AttributeName,java.lang.Class<T>
[120af7c3194ea7f36be2a09ba78a2ebf963f778d]   found: org.apache.mailet.Mail,java.lang.String,java.lang.Class<capture#1 of ? extends java.lang.Throwable>
[120af7c3194ea7f36be2a09ba78a2ebf963f778d]   reason: cannot infer type-variable(s) T
[120af7c3194ea7f36be2a09ba78a2ebf963f778d]     (argument mismatch; java.lang.String cannot be converted to org.apache.mailet.AttributeName)
[120af7c3194ea7f36be2a09ba78a2ebf963f778d] [ERROR] /james-project/mailet/standard/src/main/java/org/apache/james/transport/matchers/SentByMailet.java:[45,54] incompatible types: java.lang.String cannot be converted to org.apache.mailet.AttributeName
[120af7c3194ea7f36be2a09ba78a2ebf963f778d] [ERROR] /james-project/mailet/standard/src/main/java/org/apache/james/transport/matchers/SMTPAuthSuccessful.java:[52,54] incompatible types: java.lang.String cannot be converted to org.apache.mailet.AttributeName
[120af7c3194ea7f36be2a09ba78a2ebf963f778d] [INFO] 4 errors 
```

File: mailet/standard/src/main/java/org/apache/james/transport/mailets/AmqpForwardAttribute.java [hunk 1]
```diff
@@ -106,32 +110,40 @@ public void init() throws MailetException {
...
-        } catch (TimeoutException e) {
-            LOGGER.error("TimeoutException while writing to AMQP: {}", e.getMessage(), e);
-        } catch (AlreadyClosedException e) {
-            LOGGER.error("AlreadyClosedException while writing to AMQP: {}", e.getMessage(), e);
-        }
+        getAttributeContent(mail)
+            .ifPresent(content -> {
+                try {
+                    sendContent(content);
+                } catch (IOException e) {
```
Comment (aduprat) [inline mailet/standard/src/main/java/org/apache/james/transport/mailets/AmqpForwardAttribute.java]: I think we can group those exceptions, as the exception root cause will be in the stacktrace & they are all handled the same way.

**Comment by chibenwa** [comment]
Looks like you forgot to push before leaving the office

**Comment by chibenwa** [comment]
test this please

**Comment by hoangdat** [comment]
Build triggered. sha1 is merged.

**Comment by hoangdat** [comment]
Build started sha1 is merged.

**Comment by hoangdat** [comment]
Build finished. No test results found.

**Comment by hoangdat** [comment]
Build triggered. sha1 is merged.

**Pull request closed without merging**